template_test/Performance_template.py

```diff
@@ -0,0 +1,228 @@
+
+    """
+
+    def __init__(self, uuid, crd, full_log_path):
+        """
+        Initialize the object by reading some of the data from the CRD file and
+        by connecting to the ES server and read all results from it.
+
+        Args:
+            uuid (str): the unique uid of the test
+            crd (dict): dictionary with test parameters - the test yaml file
+                        that modify it in the test itself.
+            full_log_path (str): the path of the results files to be found
+
+        """
+        super(ResultsAnalyse, self).__init__(uuid, crd)
+
+        # The self.new_index parameter should be unique in the OCS-CI and in the ES
+        self.new_index = "testname_fullres"
+
+        self.full_log_path = full_log_path
+
+        # make sure we have connection to the elastic search server
+        self.es_connect()
+
+    def analyze_the_results(self, test_results):
+        """
+        Analyzing the test results
+
+        Args:
+            test_results (type) : description of argument
+
+        Return:
+            type : description of return value
+        """
+
+        # in this place do the in-test results analyze.
+        # this function is not mandatory.
+
+    def results_logging(self):
+        """
+        Logging relevant results to the test log
+
+        """
+        log.info("Test Results Summery:")
+
+        # Log here the summer of the test results
+
+
+@performance
+class TestClassName(PASTest):
+    """
+    Test to ...
+
+    """
+
+    def setup(self):
+        """
+        Setting up test parameters
+        """
+        log.info("Starting the test setup")
+        super(TestClassName, self).setup()
+
+        # Benchmark name need to be unique in OCS-CI
+        self.benchmark_name = "<name>"
+        self.uuid = uuid4().hex
+        self.crd_data = {
+            "spec": {
+                # values of test_user and clustername are only 'Place holders'
+                "test_user": "Homer simpson",
+                "clustername": "test_cluster",
+                "elasticsearch": {
+                    "server": config.PERF.get("production_es_server"),
+                    "port": config.PERF.get("production_es_port"),
+                    "url": f"http://{config.PERF.get('production_es_server')}:{config.PERF.get('production_es_port')}",
+                },
+            }
+        }
+        # during development use the dev ES so the data in the Production ES will be clean.
+        if self.dev_mode:
+            self.crd_data["spec"]["elasticsearch"] = {
+                "server": config.PERF.get("dev_es_server"),
+                "port": config.PERF.get("dev_es_port"),
+                "url": f"http://{config.PERF.get('dev_es_server')}:{config.PERF.get('dev_es_port')}",
+            }
+
+        # Any other test setup need to be here
+
+    def teardown(self):
+        """
+        Teardown of the test
+
+        """
+
+        # teardown operation need to be here
```

Comment:
Instead of comment can we add something like TODO:

Reply:
done